balancer/state.go

```diff
@@ -0,0 +1,102 @@
+package balancer
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/R-a-dio/valkyrie/config"
+)
+
+// Balancer represents the state of the load balancer.
+type Balancer struct {
+	config.Config
+	manager radio.ManagerService
+	relays  []*radio.Relay
+
+	serv      *http.Server
+	listeners int
+	min       float64
+	current   atomic.Value
```

Comment:
consider wrapping this field into a type with methods to set/get so that you get typechecking by the compiler

Reply:
In the new version, there is a `setCurrent` which accepts a string, so we know at compile time that only a string is being loaded into the `atomic.Value`.